### ISSUE TITLE
False negative with Layout/HashAlignment with table style

### ISSUE THREAD

**Issue by edouard-per-angusta** [rubocop/rubocop#9888]
When a key follows a longer one, the following key has a warning:

<img width="277" alt="Screenshot 2021-06-22 at 09 11 30" src="https://user-images.githubusercontent.com/77271328/122880372-3a067e00-d33a-11eb-806f-14795ca84b3c.png">

--------

## Expected behavior

There shouldn't be any warning.

## Actual behavior

There is a warning:

<img width="750" alt="Screenshot 2021-06-22 at 09 15 09" src="https://user-images.githubusercontent.com/77271328/122880519-5d312d80-d33a-11eb-9972-d04cf07d610f.png">

Running `rubocop -A` adds an extra space:

<img width="112" alt="Screenshot 2021-06-22 at 09 25 34" src="https://user-images.githubusercontent.com/77271328/122881780-d1b89c00-d33b-11eb-94c5-6a7bd0a01e3f.png">

## Steps to reproduce the problem

Create a folder with these files:

```yml
# .rubocop.yml

Layout/HashAlignment:
  EnforcedHashRocketStyle: table
  EnforcedColonStyle: table
```

```ruby
# example.rb

FOO = {
  a:  1,
  bb: 2,
  c:  3
}.freeze
```

## RuboCop version

```
$ rubocop -V
1.15.0 (using Parser 3.0.1.1, rubocop-ast 1.7.0, running on ruby 2.6.7 x86_64-darwin20)
  - rubocop-performance 1.11.3
  - rubocop-rails 2.10.1
  - rubocop-rspec 2.3.0
```

**Comment by rrosenblum** [comment]
hmm, it seems to somehow want to align with the comma rather than the value itself. This certainly seems like a bug. I've run into some other conditions recently that HashAlignment doesn't seem to be picking up. My issues have mainly been with dealing with values that appear on the next line. I haven't had too much time to dig into what's going on.

**Comment by AndreiEres** [comment]
Interesting, I can't catch the problem in `hash_alignment_spec.rb`
![image](https://user-images.githubusercontent.com/27277055/132455913-00a4dbee-8b6b-4a4d-9688-5e9bb14de043.png)

**Comment by bbatsov** [comment]
Can't reproduce on current master - a correctly table-aligned hash no longer registers a false offense under `EnforcedColonStyle: table`. Closing as fixed.
